### PR TITLE
Data model updates for MetaWorkflowRuns

### PR DESCRIPTION
**Pull request by drio18**
In this PR, we create new metadata properties on MetaWorkflows, MetaWorkflowRuns, and MetaWorkflowRun outputs (FileProcessed, QualityMetric) that are required for related changes in [foursight](https://github.com/dbmi-bgm/foursight-cgap/pull/76) and [magma](https://github.com/dbmi-bgm/magma/pull/7).

Specifically, we:

- Move MetaWorkflowRuns off of Cases and onto SampleProcessings (will handle existing Case items once merged and then delete properties on Case)
- Add properties to MWFR's output to facilitate searches on output items
- Add 2 new MWFR `final_status` options (`stopped` for manually stopped items, `quality metric failed` for those stopped due to output QC failure)
- Add properties to handle PATCHing of MWFR output files to appropriate destinations (Sample.processed_files or SampleProcessing.processed_files, currently)
- Fix a small embed API error noticed incidentally during foursight testing
- Add properties related to identifying VCFs for ingestion and files for HiGlass display, as bioinformatics is insisting on changing/having flexible file type descriptions (which kills current routes of finding these)

TODO: Merge current master and deploy to cgapwolf for bioinformatics to test pipelines with updated foursight.

### CODE REVIEW

File: src/encoded/schemas/meta_workflow.json [hunk 1]
```diff
@@ -220,15 +220,77 @@
                     },
                     "custom_pf_fields": {
                         "title": "Custom PF fields",
-                        "description": "Custom fields to be added to processed file items through Tibanna",
+                        "description": "Custom fields to be added to specified processed file items through Tibanna",
                         "type": "object",
-                        "additionalProperties": true
+                        "additionalProperties": {
```
Comment (sbreiff) [inline src/encoded/schemas/meta_workflow.json:225]: shouldn't this be `items`? `additionalProperties` is usually a bool

File: src/encoded/schemas/workflow_run.json [hunk 2]
```diff
@@ -246,6 +246,12 @@
                 "description": "one of the commands that are run for the workflow run",
                 "type": "string"
             }
+        },
+        "associated_meta_workflow_run": {
+            "title": "Associated MetaWorkflowRun",
+            "description": "The MetaWorkflowRun that produced this item",
+            "type": "string",
+            "format": "uuid"
```
Comment (sbreiff) [inline src/encoded/schemas/workflow_run.json:254]: We could have situations where a wfr is associated with more than one mwfr (for example, if a mwfr is copied and only a few final steps rerun). Will that cause problems here? Should this be an array?

File: src/encoded/types/case.py [hunk 3]
```diff
@@ -399,13 +395,16 @@ def vcf_file(self, request, sample_processing=None):
         files_processed = sp_data.get('processed_files', [])
         if not files_processed:
             return vcf_file
```
Comment (sbreiff) [inline src/encoded/types/case.py:397]: Not new in this PR, but why are we returning a blank dictionary here, especially if the output isn't a dictionary when present? Should we initialize as None instead?

File: src/encoded/schemas/workflow_run.json (hunk 2 above)
Comment (sbreiff) [reply src/encoded/schemas/workflow_run.json:254]: Maybe it would help to discuss this with Michele and Phil.

**Review by willronchetti — APPROVED** [review]
I believe these changes are structurally sound but we should note we are greatly increasing the interconnectivity of some types here and should be on the lookout for performance issues. I think hard vs. soft link can be debated in some situations but I trust your judgement on that.

**Pull request merged**